Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -911,19 +996,14 @@ IPCCommandResult ES::GetTitles(const IOCtlVRequest& request)
   if (!request.HasNumberOfValidVectors(1, 1))
     return GetDefaultReply(ES_PARAMETER_SIZE_OR_ALIGNMENT);
 
-  u32 MaxCount = Memory::Read_U32(request.in_vectors[0].address);
-  u32 Count = 0;
-  for (int i = 0; i < (int)s_title_ids.size(); i++)
+  const std::vector<u64> titles = GetInstalledTitles();
+
+  const size_t max_count = Memory::Read_U32(request.in_vectors[0].address);
+  for (size_t i = 0; i < std::min(max_count, titles.size()); i++)
   {
-    Memory::Write_U64(s_title_ids[i], request.io_vectors[0].address + i * 8);
-    INFO_LOG(IOS_ES, "IOCTL_ES_GETTITLES: %08x/%08x", (u32)(s_title_ids[i] >> 32),
-             (u32)s_title_ids[i]);
-    Count++;
-    if (Count >= MaxCount)
-      break;
+    Memory::Write_U64(titles[i], request.io_vectors[0].address + static_cast<u32>(i) * 8);
```

Comment:
[\<climits\>](http://en.cppreference.com/w/cpp/types/climits) defines `CHAR_BIT`, if this `8` corresponds to the number of bits in a byte/char.